libpod/container.go

```diff
@@ -1046,16 +1096,28 @@ func (c *Container) export(path string) error {
 
 // AddArtifact creates and writes to an artifact file for the container
 func (c *Container) AddArtifact(name string, data []byte) error {
+	if !c.valid {
+		return ErrCtrRemoved
+	}
+
 	return ioutil.WriteFile(c.getArtifactPath(name), data, 0740)
 }
 
 // GetArtifact reads the specified artifact file from the container
 func (c *Container) GetArtifact(name string) ([]byte, error) {
+	if !c.valid {
+		return nil, ErrCtrRemoved
+	}
+
 	return ioutil.ReadFile(c.getArtifactPath(name))
 }
 
 // RemoveArtifact deletes the specified artifacts file
 func (c *Container) RemoveArtifact(name string) error {
+	if !c.valid {
```

Comment:
no lock  needed here?

Reply:
We aren't directly modifying the container, just a file we store with it. We can probably rely on the OS to avoid races around removal. Might be smart to have something to prevent races while writing, though. I'll poke at that.